chainer/functions/tensor_network.py

```diff
@@ -0,0 +1,264 @@
+import math
+
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _as_vec(x):
+    return x.reshape(x.size)
+
+
+def _as_mat(x):
+    return x.reshape(x.shape[0], x.size // x.shape[0])
+
+
+def _empty_like(x):
```

Comment:
There are at least three `_as_mat` functions in Chainer (tensor_network, linear, matmul). I will create as another PR that unify these functions if such change is preferable.